### ISSUE TITLE
Boolean not handled consistently when parsing parameter into controller method

### ISSUE THREAD

**Issue by raphaelbauer** [ninjaframework/ninja#557]
Originally from https://github.com/ninjaframework/ninja/pull/554#issuecomment-273621764

**Comment by raphaelbauer** [comment]
jjlauer says:
hey @raphaelbauer was using beta2 and had a couple questions. I turned strict mode on and my app gets a bad request exception (as expected) if the parameter is not provided. Booleans don't work like i thought though. Seems like if the value doesn't match "true" then its just false. I was wondering if we should be more strict and actually require it to be true or false otherwise null? Integers and Longs do work as expected, btw. If you try to set it to a string, you trigger a bad request exception.

Only other feedback is that it'd be nice to know on the console/log file what parameter failed to convert and triggered the bad request exception. Right nows its a tad hard to know what param triggered it.

Optionals are awesome to use and make a lot of sense. Thank you for cranking this out!

**Comment by raphaelbauer** [comment]
rbauer answers:
Ah sorry... I think I misread that...
You mean that we always get "false" as boolean value as long as it is not null... right?

This comes from:
Boolean's
```
    public static boolean parseBoolean(String s) {
        return ((s != null) && s.equalsIgnoreCase("true"));
    }
```

which we use here:

```
public static class BooleanParamParser implements ParamParser<Boolean> {
        @Override
        public Boolean parseParameter(String field, String parameterValue, Validation validation) {
            if (parameterValue == null || parameterValue.isEmpty() || validation.hasFieldViolation(field)) {
                return null;
            } else {
                return Boolean.parseBoolean(parameterValue);
            }
        }

        @Override
        public Class<Boolean> getParsedType() {
            return Boolean.class;
        }
    }
```

That looks like a bug. And it is not really related to Optional, but to the way we handle Booleans in controllers... I think that is wrong...

I'd propose to fix that in the BooleanParamParser (and also PrimitiveBooleanParamParser)...  (but it may be a breaking change for some).

**Comment by raphaelbauer** [comment]
jjlauer:
Yes, looks like you found it.  I think just Boolean behaves that way.
You're suggested fix sounds right.  I'd expect a case insenstive "true" and
"false" to work, but something like "blah" to be null (rather than false).

Do you think it'd be possible to include a little bit of info about which
parameter caused the BadRequestException as well?  It would help narrow
down what happened IMHO.

**Comment by raphaelbauer** [comment]
I'll create testcases and a fix shorty...

**Comment by raphaelbauer** [comment]
Could you check https://github.com/ninjaframework/ninja/pull/558

Feedback well appreciated! Thanks :)

**Comment by raphaelbauer** [comment]
Fixed by #558
